File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string arr[10] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

long long solution(string numbers) {
    long long answer = 0;

    for (int i = 0; i < 10; i++) {
        while(numbers.find(arr[i]) != string::npos) {
            numbers.replace(numbers.find(arr[i]), arr[i].length(), to_string(i));
        }
    }

    answer = stoll(numbers);

    return answer;
}

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.swift
import Foundation

func solution(_ numbers:String) -> Int64 {
    var result = numbers
    var num = 0
    
    for number in ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"] {
        while result.contains(number) {
            result = result.replacingOccurrences(of: number, with: String(num))
        }
        num += 1
    }

    return Int64(result)!
}